docs/package-yanking-policies.md

```diff
@@ -0,0 +1,156 @@
+
+This paper discusses the components of modern package deletion policies. It is not meant to be a sample deletion policy, but a discussion of things to consider when crafting such a policy. 
+
+## Deletion policy considerations
+
+If we consider a hypothetical package deletion policy, it would exist on a spectrum somewhere between never allowing packages to be deleted to allowing packages to be deleted without any restrictions. Both ends of this spectrum are impractical. 
+
+* Never allowing for package deletion could result in the publishing of malicious, abusive, illegal, or copywritten content without any recourse.
+* Allowing for unrestricted package deletion could result in build failures like those seen in the `left-pad` incident, malicious packages snapping up a recently vacated namespace, and other undesireable behavior. 
+
+So when crafting a deletion policy, we want to stay away from the two extremes, but what needs to be considered?
+
+Whether or not a package manager allows for users to delete packages in some cases, it should allow for the package manager maintainers to manually intervene and delete packages. This allows the maintainer team to protect the ecosystem from malicious, abusive, illegal, or copywritten content. 
+
+But what if a user wants to delete a package they published? Although it's possible to require such cases to go to the maintainer team to be manually reviewed and removed, that is not a very efficient policy. Such a policy would place a burden on the (already likely overworked) maintainer team, and the ecosystem would be cluttered with the unwanted packages until they are manually removed. 
+
+Allowing a user to delete their package in certain cases is therefore good for the user, the package manager maintainer team, and the package ecosystem at large. But what factors need to be considered to avoid a `left-pad` style incident? 
+
+## Ecosystem impact
+
+A modern package deletion policy aims to minimize the impact a given deletion could have on the ecosystem, while granting users some flexibilty with their packages. 
+
+The following considerations are generally used when determining whether a package is a good candidate for deletion, although specific limits and criteria will vary between package managers. 
+
+### Time
+
+How long ago was the package published? Although time since publication is not a perfect metric for ecosystem impact (maybe a package has been around for years but only has a handful of users), there are a number of use cases where this metric is very relevant. For example, the user could have:
+
+* Published the wrong package
+* Published the package with an unintended name (or a typo)
+* Published a test package to understand and experience the package publishing process
+
+In these cases, it is very useful for the package manager to have a no hassle deletion policy as long as the package has been published for less than a certain amount of time. 
```

Comment:
```suggestion
In these cases, it is very useful for the package manager to have a no hassle deletion policy as long as the package has been published for less than a certain amount of time (e.g., 7 days). 
```

Reply:
Give an example. Feel free to change the example, but giving an example would help clarify things.